### DISCUSSION TITLE
About the counter objects's position

### DISCUSSION THREAD

**Discussion by SulymanHossain** [AstroBridge/BDLensing#228] [Q&A] [answered]
Currently, I'm running PSO for the system: DESIJ1539+1650. My model is creating the Arcs well. But It can't correctly locate the counter object. How can I fix its location by hard coding?

**Comment by SulymanHossain** [comment]
@ajshajib

**Comment by ajshajib** [comment, marked as answer]
You first need to find the position of the counter image in RA and Dec coordinates. Then, the `center_x` and `center_y` positions of the corresponding source components (Sersic, shapelets, etc.) needs to be initiated at those values with a tight upper and lower bound. In `kwargs_constraints`, you need to include a keyword `"image_plane_source_list"`, whose value is a list of booleans with the same length as the source model list. For example, if your source model list is `["SERSIC_ELLIPSE", "SHAPELETS"]`, then it should be set like this:

```
kwargs_constraints = {
    ...
    "image_plane_source_list": [True, True],
    ...
}
```

**Reply by SulymanHossain** [reply]
Thanks, it worked.
